sql/pgwire/encoding.go

```diff
@@ -78,7 +59,11 @@ func (b *readBuffer) readUntypedMsg(rd io.Reader) error {
 			size, maxMessageSize)
 	}
 
-	b.reset(size)
+	if len(b.msgBuf) < size {
```

Comment:
Is it safe to reuse msgBuf? My memory says no, but I'm not looking at the
code.

On Friday, November 27, 2015, Tobias Schottdorf <notifications@github.com>
wrote:

> In sql/pgwire/encoding.go
> <https://github.com/cockroachdb/cockroach/pull/3240#discussion_r46068009>:
>
> > @@ -78,7 +59,11 @@ func (b *readBuffer) readUntypedMsg(rd io.Reader) error {
> >  			size, maxMessageSize)
> >  	}
> >
> > -	b.reset(size)
> > +	if len(b.msgBuf) < size {
>
> gotcha.
> … <#1514a79d41d72d45_>
> On Fri, Nov 27, 2015 at 2:38 PM, Tamir Duberstein <
> notifications@github.com
> <javascript:_e(%7B%7D,'cvml','notifications@github.com');>> wrote: In
> sql/pgwire/encoding.go <#3240 (comment)
> <https://github.com/cockroachdb/cockroach/pull/3240#discussion_r46067833>>:
> > @@ -78,7 +59,11 @@ func (b *readBuffer) readUntypedMsg(rd io.Reader)
> error { > size, maxMessageSize) > } > > - b.reset(size) > + if
> len(b.msgBuf) < size { not really; If you look at all the methods on
> readBuffer, they advance the slice each time something is read - without
> msgBuf it would be impossible to recover the front of the buffer. — Reply
> to this email directly or view it on GitHub <
> https://github.com/cockroachdb/cockroach/pull/3240/files#r46067833>.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/cockroachdb/cockroach/pull/3240/files#r46068009>.
>